src/markov/markov_approx.jl

```diff
@@ -219,3 +223,567 @@ function estimate_mc_discrete{T}(X::Vector{T})
+    end
+
+    X = C*D + repmat(mu, 1, Nm^M) # map grids back to original space
+
+    M != 1 || (return MarkovChain(P, vec(X)))
+    return MarkovChain(P, [X[:, i] for i in 1:Nm^M])
+end
+
+"""
+
+return the size of scalar as (1, 1)
+
+##### Arguments
+
+- `x::Real` : Scalar value
+
+##### Return
+
+- `(1, 1)`
+
+"""
+Base.size(x::Real) = (1, 1)
+
+"""
+
+return standerdized AR(1) representation
+
+##### Arguments
+
+- `b::Real` : constant term
+- `B::Real` : impact coefficient
+- `Psi::Real` : variance of innovation
+- `M::Integer == 1` : must be one since the function is for AR(1)
+
+##### Returns
+
+- `A::Real` : impact coefficient of standardized AR(1) process
+- `C::Real` : standard deviation of the innovation
+- `mu::Real` : mean of the standardized AR(1) process
+- `Sigma::Real` : variance of the standardized AR(1) process
+
+"""
+function standardize_var(b::Real, B::Real, Psi::Real, M::Integer)
+    C = sqrt(Psi)
+    A = B
+    mu = [b/(1-B)] # mean of the process
+    Sigma = 1/(1-B^2) #
+    return A, C, mu, Sigma
+end
+
+"""
+
+return standerdized VAR(1) representation
+
+##### Arguments
+
+- `b::Vector` : (M x 1) constant term vector
+- `B::Matrix` : (M x M) matrix of impact coefficients
+- `Psi::Matrix` : (M x M) variance-covariance matrix of innovations
+- `M::Intger` : number of variables of the VAR(1) model
+
+##### Returns
+
+- `A::Matirx` : impact coefficients of standardized VAR(1) process
+- `C::Matrix` : variance-covariance matrix of standardized model innovations
+- `mu::Vector` : mean of the standardized VAR(1) process
+- `Sigma::Matrix` : variance-covariance matrix of the standardized VAR(1) process
+
+"""
+function standardize_var(b::Vector, B::Matrix, Psi::Matrix, M::Integer)
+    C1 = cholfact(Psi)[:L]
+    mu = ((eye(M)-B)\eye(M))*b
+    A1 = C1\(B*C1)
+    # unconditional variance
+    Sigma1 = reshape(((eye(M^2)-kron(A1,A1))\eye(M^2))*vec(eye(M)),M,M)
+    U, _ = min_var_trace(Sigma1)
+    A = U'*A1*U
+    Sigma = U'*Sigma1*U
+    C = C1*U
+    return A, C, mu, Sigma
+end
+
+"""
+construct prior guess for evenly spaced grid method
+
+##### Arguments
+
+- `condMean::Vector` : conditional Mean of each variable
+- `Nm::Integer` : number of grid points
+- `y1D::Matrix` : grid of variable
+- `method::Even` : method for grid making
+
+"""
+construct_prior_guess(condMean::Vector, Nm::Integer, y1D::Matrix, method::Even) =
+    pdf.(Normal.(repmat(condMean, 1, Nm), 1), y1D)
+
+"""
+
+construct one-dimensional grid of states
+
+##### Argument
+
+- `Sigma::ScalarOrArray` :
+- `Nm::Integer` : number of grid points
+- `M::Integer` : number of variables (M=1 corresponds to AR(1))
+- `nSigmas::Real` : number of standard error determining end points of grid
+- `method::Even` : method for grid making
+
+##### Return
+
+- `y1D` : (M x Nm) matrix of variable grid
+
+"""
+function construct_1D_grid(Sigma::ScalarOrArray, Nm::Integer,
+                           M::Integer, nSigmas::Real, method::Even)
+    if M == 1
```

Comment:
I _believe_ we can drop the conditional on `M` here by doing 

```julia
minSigmas = sqrt(minimum(eigfact(Sigma).values))
```

`eigfact` will work for both scalars and matrices